File: news-mobile/lib/function/scaffold.dart
import 'package:flutter/material.dart';
import 'package:mobile/clients/news_client.dart';
import 'package:mobile/function/widget.dart';

Scaffold buildScaffold(
    NewsClient client, BuildContext context, Map<String, dynamic> attrs) {
  Widget appBar = buildWidget(
    client,
    context,
    attrs['appBar']['type'],
    attrs['appBar']['attrs'],
  );
  Widget drawer = attrs['drawer'] != null
      ? buildWidget(
          client,
          context,
          attrs['drawer']['type'],
          attrs['drawer']['attrs'],
        )
      : null;
  Widget body = buildWidget(
    client,
    context,
    attrs['body']['type'],
    attrs['body']['attrs'],
  );
  Widget floatingActionButton = buildWidget(
    client,
    context,
    attrs['floatingActionButton']['type'],
    attrs['floatingActionButton']['attrs'],
  );
  return Scaffold(
    appBar: appBar,
    drawer: drawer,
    body: body,
    floatingActionButton: floatingActionButton,
  );
}

File: news-flutter/src/main/java/com/angkorteam/news/flutter/widget/Scaffold.java
package com.angkorteam.news.flutter.widget;

import com.angkorteam.news.flutter.Page;
import com.angkorteam.news.flutter.common.Entry;
import com.angkorteam.news.flutter.common.Widget;
import org.jooq.DSLContext;

public class Scaffold extends Widget<ScaffoldAttribute> {

    public Scaffold(Entry<ScaffoldAttribute, Object>... attrs) {
        super(attrs);
    }

    public static Entry<ScaffoldAttribute, Object> appBar(Widget appBar) {
        return new Entry<>(ScaffoldAttribute.appBar, appBar);
    }

    public static Entry<ScaffoldAttribute, Object> body(Widget body) {
        return new Entry<>(ScaffoldAttribute.body, body);
    }

    public static Entry<ScaffoldAttribute, Object> floatingActionButton(Widget floatingActionButton) {
        return new Entry<>(ScaffoldAttribute.floatingActionButton, floatingActionButton);
    }

    @Override
    public void store(DSLContext context, Page page) {
        store(context, page, ScaffoldAttribute.values());
        super.store(context, page);
    }

}
